### PR TITLE
Core/Authserver: Fix uninitialized variable, fixes changing realms aFter logging in to world

### LINKED ISSUE #11388: Unable to change realm after entering to one of them.
### Current Behaviour

Unable to change realm after entering to one of them. But if you not entered to world, you can choose and enter to one of the listed realms 

See video

https://discord.com/channels/217589275766685707/284323424032129024/963712657146388490

### Expected Blizzlike Behaviour

You can choose and enter to one of the listed realms in any time if went out from the other realms

### Source

_No response_

### Steps to reproduce the problem

1. start one server with ID 1 and port 8585
2. start second server with ID 2 and port 8586
3. add realms to realmlist table of auth database
4. log in to auth section
5. now you can choose the server and switch between them
6. enter to one of them and leave it to previous screen
7. now you can't switch the servers.
8. If you completelly log out and login again - you will be able to switch servers.

### Extra Notes

_No response_

### AC rev. hash/commit

AzerothCore rev. e13f4271897e+ 2022-04-11 17:04:42 +0000

### Operating system

Win7 x64

### Custom changes or Modules

none

### PR DESCRIPTION
**Pull request by r0m1ntik**
## Issues Addressed:

- Closes #11388 

## SOURCE:

[Pick up from TC](https://github.com/TrinityCore/TrinityCore/commit/4ec52fb160c9bd5ab43a305756e4e4066c857994)
Co-Authored-By: Shauren <[shauren.trinity@gmail.com](mailto:shauren.trinity@gmail.com)>

## Tests Performed:

- Need to test

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Kitzunu — APPROVED** [review]

File: src/server/authserver/Server/AuthSession.cpp [hunk 1]
```diff
@@ -847,7 +847,7 @@ bool AuthSession::VerifyVersion(uint8 const* a, int32 aLength, Acore::Crypto::SH
     if (!sConfigMgr->GetOption<bool>("StrictVersionCheck", false))
         return true;
 
-    Acore::Crypto::SHA1::Digest zeros;
+    Acore::Crypto::SHA1::Digest zeros = { };
     Acore::Crypto::SHA1::Digest const* versionHash = nullptr;
```
Comment (IntelligentQuantum) [inline src/server/authserver/Server/AuthSession.cpp:851]: Acore::Crypto::SHA1::Digest zeros{};
    Acore::Crypto::SHA1::Digest const* versionHash{ nullptr };

Comment (r0m1ntik) [reply src/server/authserver/Server/AuthSession.cpp:851]: Thx :)

Comment (Winfidonarleyan) [reply src/server/authserver/Server/AuthSession.cpp]: https://github.com/WarheadCore/WarheadBand/commit/a7ca2d4c39a761cc13fc7cdc4ac771fe2163f269 :D

Comment (IntelligentQuantum) [reply src/server/authserver/Server/AuthSession.cpp]: yes but https://www.modernescpp.com/index.php/initialization xD

**Review by IntelligentQuantum — APPROVED** [review]

**Pull request merged**